Sources/MongoSwift/MongoCollection+Read.swift

```diff
@@ -88,7 +88,39 @@ extension MongoCollection {
         options: AggregateOptions? = nil,
         session: ClientSession? = nil
     ) -> EventLoopFuture<MongoCursor<BSONDocument>> {
-        let operation = AggregateOperation(collection: self, pipeline: pipeline, options: options)
+        self.aggregate(pipeline, options: options, session: session, withOutputType: BSONDocument.self)
+    }
+
+    /**
+     * Runs an aggregation framework pipeline against this collection.
+     * Associates the `Codable` type `OutputType` with the output given by the `AggregateOperation`.
+     *
+     * - Parameters:
+     *   - pipeline: an `[Document]` containing the pipeline of aggregation operations to perform
+     *   - options: Optional `AggregateOptions` to use when executing the command
+     *   - session: Optional `ClientSession` to use when executing this command
+     *   - withOutputType: the type that the output of the aggregation operation will be decoded to
```

Comment:
done, should i make this clearer in the first part of the documentation also? or is this clear enough

Reply:
Maybe update the last half of the first part of the docstring to say the following:
>"Associates the specified `Codable` type `OutputType` with the returned `MongoCursor`" 

or something like that. It's pretty similar to what we say for [change streams](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/MongoCollection%2BChangeStreams.swift#L86): 

>"Associates the specified `Codable` type `T` with the returned `ChangeStream`"

This does make me realize the change stream one should be updated to say `EventType` instead of `T` (you don't need to fix that here).

Reply:
ok sounds good